core/src/main/java/org/testcontainers/containers/PortForwardingContainer.java

```diff
@@ -0,0 +1,69 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.model.ContainerNetwork;
+import com.trilead.ssh2.Connection;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+public enum PortForwardingContainer {
+    INSTANCE;
+
+    private GenericContainer container;
+
+    private final Set<Integer> exposedPorts = Collections.newSetFromMap(new ConcurrentHashMap<>());
+
+    @Getter(value = AccessLevel.PRIVATE, lazy = true)
+    private final Connection sshConnection = createSSHSession();
+
+    @SneakyThrows
+    private Connection createSSHSession() {
+        String password = UUID.randomUUID().toString();
+        container = new GenericContainer<>(TestcontainersConfiguration.getInstance().getSSHdImage())
+            .withExposedPorts(22)
+            .withEnv("PASSWORD", password)
+            .withCommand(
+                "sh",
+                "-c",
+                // Disable ipv6 & Make it listen on all interfaces, not just localhost
+                "echo \"root:$PASSWORD\" | chpasswd && /usr/sbin/sshd -D -o PermitRootLogin=yes -o AddressFamily=inet -o GatewayPorts=yes"
+            );
+        container.start();
+
+        Connection connection = new Connection(container.getContainerIpAddress(), container.getMappedPort(22));
+
+        connection.setTCPNoDelay(true);
+        connection.connect(
+            (hostname, port, serverHostKeyAlgorithm, serverHostKey) -> true,
+            (int) Duration.ofSeconds(30).toMillis(),
+            (int) Duration.ofSeconds(30).toMillis()
+        );
+
+        if (!connection.authenticateWithPassword("root", password)) {
+            throw new IllegalStateException("Authentication failed.");
+        }
+
+        return connection;
+    }
+
+    @SneakyThrows
+    public void exposeHostPort(int port) {
+        if (exposedPorts.add(port)) {
+            getSshConnection().requestRemotePortForwarding("", port, "localhost", port);
+        }
+    }
+
+    Optional<ContainerNetwork> getNetwork() {
+        return Optional.ofNullable(container)
```

Comment:
I think we have the same exact code multiple times in our code base, so we could add a utility method for this? (not in this PR).

`Optional<ContainerNetwork> getFirstNetwork()` in `GenericContainer?`

Reply:
this code doesn't have any logic in it (and just calls the getters), but exposing it as public API (in Testcontainers) makes me feel uneasy, I would rather not do that